spring-integration-core/src/test/java/org/springframework/integration/gateway/GatewayInterfaceTests.java

```diff
@@ -683,7 +671,7 @@ public interface ExecGateway {
 		Future<Thread> test1(Thread caller);
 
 		@Gateway(requestChannel = "gatewayThreadChannel")
-		ListenableFuture<Thread> test2(Thread caller);
+		CompletableFuture<Thread> test2(Thread caller);
```

Comment:
Shouldn't we leave some of these tests as Listenable, to make sure they're not broken?